### PR TITLE
Add `scarb-doc`-generated corelib docs to GitHub Pages

### LINKED ISSUE #1381: Automatic docs generation for `corelib`

### PR DESCRIPTION
**Pull request by DelevoXDG**
Closes #1381

Preview: 
- https://delevoxdg.github.io/scarb/corelib/index.html
- https://delevoxdg.github.io/scarb/docs

### CODE REVIEW

**Comment by piotmag769** [comment]
No structs seem to have a signature in ur demo link. @wawel37 is that a scarb-doc bug?

**Comment by wawel37** [comment]
> No structs seem to have a signature in ur demo link. @wawel37 is that a scarb-doc bug?

Hmm, strange. @DelevoXDG , what version have you used to generate those docs? Was it some tagged version of Scarb or just the current main branch scarb? Because as I'm testing locally, all of the structs got the signatures (main branch version of scarb).

**Comment by wawel37** [comment]
> Member

Hmmm, strange. @DelevoXDG , what version of scarb is used by default with `uses: software-mansion/setup-scarb@v1`? It's either because of some older version of scarb which doesn't have the struct signatures implemented yet, or just a bug in scarb-doc. It's strange because as I'm testing it right now (scarb main branch build), everything work just fine

File: .github/workflows/website-deploy.yml [hunk 1]
```diff
@@ -16,8 +16,49 @@ concurrency:
...
+  build-corelib-docs:
+    runs-on: ubuntu-latest
+    env:
+      MDBOOK_VERSION: 0.4.31
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: nightly
```
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply .github/workflows/website-deploy.yml:28]: Good catch. I suppose better docs should come with the next version of Scarb then.

@wawel37 @piotmag769

File: .github/workflows/website-deploy.yml [hunk 2]
```diff
@@ -16,8 +16,47 @@ concurrency:
...
+      - uses: software-mansion/setup-scarb@v1
+      - name: Install mdBook
+        run: cargo install --version ${MDBOOK_VERSION} mdbook
+      - name: Init Scarb to cache corelib
+        id: scarb-init
+        run: |
+          mkdir -p $HOME/pkg
+          pushd $HOME/pkg
+          scarb init --no-vcs
+          echo "CAIRO_VERSION_PATH=$(scarb metadata --format-version 1 | jq -r '.app_version_info.cairo | if .commit_info != null then .commit_info.short_commit_hash else "v\(.version)" end')" >> $GITHUB_OUTPUT
```
Comment (maciektr) [inline .github/workflows/website-deploy.yml:35]: Hmm, why commit hash? Shouldn't we always use version?

File: .github/workflows/website-deploy.yml [hunk 3]
```diff
@@ -16,8 +16,47 @@ concurrency:
...
+        env:
+          SCARB_INIT_TEST_RUNNER: cairo-test
+      - name: Extract corelib from Scarb cache
+        run: |
+          mkdir -p core
+          cp -r $(scarb cache path)/registry/std/${{ steps.scarb-init.outputs.CAIRO_VERSION_PATH }}/core/* ./core
```
Comment (maciektr) [inline .github/workflows/website-deploy.yml:41]: This can be read from metadata directly though, no need to construct the path here.

**Review by maciektr — COMMENTED** [review]
👍 
Two tiny comments attached below.

**Review by wawel37 — APPROVED** [review]

**Review by maciektr — APPROVED** [review]

File: website/.vitepress/config.mjs [hunk 4]
```diff
@@ -62,6 +62,7 @@ const sidebar = {
         p("Scripts", "/docs/reference/scripts"),
         p("Specifying dependencies", "/docs/reference/specifying-dependencies"),
         p("Targets", "/docs/reference/targets"),
+        p("Corelib documentation", "https://docs.swmansion.com/scarb/corelib/index.html"),
```
Comment (maciektr) [inline website/.vitepress/config.mjs:65]:
```suggestion
        p(
          "Corelib documentation",
          "https://docs.swmansion.com/scarb/corelib/index.html",
        ),
```

**Pull request merged**